monkey/infection_monkey/i_puppet.py

```diff
@@ -0,0 +1,88 @@
+import abc
+import threading
+from collections import namedtuple
+from enum import Enum
+from typing import Dict, Optional, Tuple
+
+
+class PortStatus(Enum):
+    OPEN = 1
+    CLOSED = 2
+
+
+PortScanData = namedtuple("PortScanData", ["port", "status", "banner", "service"])
+
+
+class IPuppet(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def run_sys_info_collector(self, name: str) -> Dict:
+        """
+        Runs a system info collector
+        :param str name: The name of the system info collector to run
+        :return: A dictionary containing the information collected from the system
+        :rtype: Dict
```

Comment:
Is this really needed? We already have type annotations.

Reply:
Yeah. This is used for generating documentation. We're not doing it at the moment, but we should in the future. We don't need to add these kinds of comments for everything, but we really should have them for our interfaces.